v2/observer/jobs_test.go

```diff
@@ -495,6 +495,39 @@ func TestDeleteJobResources(t *testing.T) {
 	}
 }
 
+func TestSyncDeletedJobPod(t *testing.T) {
+	const testNamespace = "foo"
+	const testPodName = "bar"
+	observer := &observer{
+		deletingPodsSet: map[string]struct{}{
+			namespacedPodName(testNamespace, testPodName): {},
+		},
+		syncMu: &sync.Mutex{},
+		jobsClient: &coreTesting.MockJobsClient{
+			UpdateStatusFn: func(
+				context.Context,
+				string,
+				string,
+				core.JobStatus,
+			) error {
+				return nil
```

Comment:
Not sure if overkill... do we want to assert on expected status (ABORTED) here?

Similarly, do we want to add assertions on strings passed in to `CleanupFn`?  

(If either/both are added, these comments apply to the worker pod test as well.)

Reply:
> Not sure if overkill.

Not sure either, but happy to add.

Reply:
Ok, might as well add when convenient.  Thanks!

Reply:
All set.